### PR TITLE
[qa_automation] Use virtio console in regression testing

### PR DESCRIPTION
**Pull request by jtzhao**
Tested virtio console on our local openQA server. It works properly.

http://147.2.207.102/tests/870#

### CODE REVIEW

File: tests/qa_automation/qa_run.pm [hunk 1]
```diff
@@ -50,7 +50,12 @@ sub system_status {
 sub system_login {
     my $self = shift;
     $self->wait_boot;
-    select_console('root-console');
+    if (get_var('VIRTIO_CONSOLE')) {
+        select_console('root-virtio-terminal');
+    }
+    else {
+        select_console('root-console');
+    }
```
Comment (okurz) [inline tests/qa_automation/qa_run.pm]: I guess that is better handled more low-level. Based on the variable better set a root console variable. Or do you really nowhere else again need to select the virtio console then?

Comment (mimi1vx) [inline tests/qa_automation/qa_run.pm]: @okurz where ? in this test are only two places where is selected console, and second isn't needed :) Plus for this test is virtio_console good performance optimisation, unfortunately this breaks SLE12GA/SLE12SP1 usage and needs wait for pr#2820

Comment (okurz) [inline tests/qa_automation/qa_run.pm]: then it's fine.

**Comment by jtzhao** [comment]
I see. Let's wait for #2820 . After that, I'll test it again on SLE12 and SLE12SP1.

**Comment by mimi1vx** [comment]
@jtzhao  thanks:)

File: tests/qa_automation/qa_run.pm [hunk 2]
```diff
@@ -209,6 +214,9 @@ sub run() {
     my $junit_type = $self->junit_type();
     assert_script_run("/usr/share/qa/qaset/bin/junit_xml_gen.py -n '$junit_type' -d -o /tmp/junit.xml /var/log/qaset");
     parse_junit_log("/tmp/junit.xml");
+
+    # Switch back to root-console
+    select_console('root-console');
```
Comment (asmorodskyi) [inline tests/qa_automation/qa_run.pm]: am I missing something or this step is relevant only for case when it is 'VIRTIO_CONSOLE' ?

in this case please add if (get_var('VIRTIO_CONSOLE')) so we won't waste time on selecting something which is already selected

Comment (okurz) [inline tests/qa_automation/qa_run.pm]: It does not harm to re-select the current console. We also do this all the time, e.g. when starting a test module which is executing a `select_console` call in the beginning to be independant of previous modules. I would even consider this good style. If you see a problem with that approach we should rather within the backend make sure it's a noop.

Comment (asmorodskyi) [inline tests/qa_automation/qa_run.pm]: good idea. then merging this one

**Pull request merged**